crates/iota-node/src/lib.rs

```diff
@@ -1752,8 +1773,24 @@ impl IotaNode {
                 guard.as_ref().map(|handle| {
                     let tx = handle.checkpoint_summary_broadcaster().clone();
                     Box::new(move |summary: &CertifiedCheckpointSummary| {
-                        if let Err(e) = tx.send(summary) {
-                            warn!("Failed to send checkpoint summary: {e}");
+                        if tx.receiver_count() > 0 {
+                            match tx.send(summary) {
```

Comment:
A receiver might have unsubscribed in between these two lines. Why not let `tx.send` handle `receiver_count` itself?

Reply:
I implied that the design is not great:

1. it exposes and makes use of internals (`receiver_count`),
2. `send` is a method of `CheckpointSummaryBroadcaster` trait while `receiver_count` is a method of `GrpcCheckpointSummaryBroadcaster` type.

Arguably, a better approach would be to move sending and tracing stuff into a separate method (eg. `send_traced` or `send` itself) so that the call site is not bloated with not-so-relevant boilerplate code.